paddle/fluid/operators/matmul_op.cc

```diff
@@ -409,7 +409,45 @@ class MatMulOp : public framework::OperatorWithKernel {
     if (dim_out.empty()) {
       dim_out = {1};
     }
-    context->SetOutputDim("Out", framework::make_ddim(dim_out));
+
+    framework::DDim ddim_out = framework::make_ddim(dim_out);
+
+#ifdef PADDLE_WITH_MKLDNN
+    //  if mkldnn matmul+transpose+reshape fuse activated
+    auto reshape_out =
+        context->Attrs().Get<std::vector<int>>("fused_reshape_Out");
+    auto transpose_out =
+        context->Attrs().Get<std::vector<int>>("fused_transpose_Out");
+
+    if (!reshape_out.empty() && !transpose_out.empty()) {
+      auto reshape_out_size = reshape_out.size();
+      auto transpose_out_size = transpose_out.size();
+      PADDLE_ENFORCE_EQ(transpose_out_size, 4,
+                        platform::errors::InvalidArgument(
+                            "transpose_out_size supported rank is 4, "
+                            "received %d",
+                            transpose_out_size));
+      bool supported_transpose_axis =
+          (transpose_out.at(0) == 0 && transpose_out.at(1) == 2 &&
```

Comment:
IMO if we check with enforce for supported transpose axis and reshape out size here, we should check these conditions in the fuse pass before executing it (now we apply the fuse unconditionally). With the current version, it should work fine with the Ernie model, but can lead to failure with some other model.

Reply:
Done